avocado/core/mux.py

```diff
@@ -29,6 +36,152 @@
 REMOVE_VALUE = 1
 
 
+class MuxTree(object):
+
+    """
+    Object representing part of the tree from the root to leaves or another
+    multiplex domain. Recursively it creates multiplexed variants of the full
+    tree.
+    """
+
+    def __init__(self, root):
+        """
+        :param root: Root of this tree slice
+        """
+        self.pools = []
```

Comment:
Well the change is atomic, because the information about `root` is now stored in the `MuxPlugin` rather than in each `MuxTree`. I mean it would be possible to first move it here and then introduce the `MuxPlugin`, but with so many changes I thought this little change is understandable. Anyway if you insist I can rework that (causing yet another delay).